src/Build/BackEnd/Components/SdkResolution/SdkResolverLoadingStrategy.cs

```diff
@@ -0,0 +1,74 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using Microsoft.Build.BackEnd.Logging;
+using Microsoft.Build.Construction;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Shared;
+
+namespace Microsoft.Build.BackEnd
+{
+    internal class SdkResolverLoadingStrategy
+    {
+        internal virtual IList<SdkResolver> LoadResolvers(ILoggingService logger, BuildEventContext buildEventContext,
+            ElementLocation location)
+        {
+            // Always add the default resolver
+            var resolvers = new List<SdkResolver> {new DefaultSdkResolver()};
```

Comment:
Should the default resolver only get added when there are no explicit resolvers set?

If the default is there all the time, it means it might work when the build should actually fail. Maybe the `SdkResult` should contain a `ShouldFailTheBuild` member.